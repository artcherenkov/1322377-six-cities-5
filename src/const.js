export const HousingType = {
  APARTMENT: `apartment`,
  ROOM: `room`,
  HOUSE: `house`,
  HOTEL: `hotel`
};

export const Cities = {
  PARIS: `PARIS`,
  COLOGNE: `COLOGNE`,
  BRUSSELS: `BRUSSELS`,
  AMSTERDAM: `AMSTERDAM`,
  HAMBURG: `HAMBURG`,
  DUSSELDORF: `DUSSELDORF`
};

export const MapType = {
  PROPERTY: `property__map`,
  CITIES: `cities__map`
};

export const OffersListType = {
  NEAR: `near-places__list`,
  CITIES: `cities__places-list tabs__content`
};

export const SortType = {
  DEFAULT: `DEFAULT`,
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};
