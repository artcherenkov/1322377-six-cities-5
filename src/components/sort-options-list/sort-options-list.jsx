import React from "react";
import PropTypes from 'prop-types';
import {nanoid} from "nanoid";

import {SortType} from "../../const";

export const SortOptionsList = (props) => {
  const {currentSortType, onSortTypeChange, isOpened, onRollupToggle} = props;
  return (
    <ul className={`places__options places__options--custom ${isOpened ? `places__options--opened` : ``}`} onClick={onRollupToggle}>
      {Object.values(SortType).map((sortType) => {
        if (sortType !== SortType.DEFAULT) {
          return (
            <li className={`places__option ${sortType === currentSortType ? `places__option--active` : ``}`}
              tabIndex="0"
              key={nanoid()}
              onClick={(evt) => {
                evt.preventDefault();
                onSortTypeChange(sortType);
              }}
            >{sortType}</li>
          );
        }
        return ``;
      })}
    </ul>
  );
};

SortOptionsList.propTypes = {
  currentSortType: PropTypes.string.isRequired,
  onSortTypeChange: PropTypes.func.isRequired,
  isOpened: PropTypes.bool.isRequired,
  onRollupToggle: PropTypes.func.isRequired
};
