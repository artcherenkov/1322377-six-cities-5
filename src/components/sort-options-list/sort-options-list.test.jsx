import React from 'react';
import renderer from 'react-test-renderer';

import {SortOptionsList} from './sort-options-list';
import {SortType} from "../../const";

it(`SortOptionsList is rendered correctly`, () => {
  const tree = renderer.create(
      <SortOptionsList
        currentSortType={SortType.DEFAULT}
        isOpened={true}
        onRollupToggle={() => {}}
        onSortTypeChange={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
