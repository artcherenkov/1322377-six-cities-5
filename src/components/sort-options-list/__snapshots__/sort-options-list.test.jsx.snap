// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortOptionsList is rendered correctly 1`] = `
<ul
  className="places__options places__options--custom places__options--opened"
  onClick={[Function]}
>
  
  <li
    className="places__option"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
  </li>
  <li
    className="places__option"
    onClick={[Function]}
    tabIndex="0"
  >
    Price: low to high
  </li>
  <li
    className="places__option"
    onClick={[Function]}
    tabIndex="0"
  >
    Price: high to low
  </li>
  <li
    className="places__option"
    onClick={[Function]}
    tabIndex="0"
  >
    Top rated first
  </li>
</ul>
`;
